README.md

```diff
@@ -3,7 +3,7 @@
 This plugin serves as a starting point for writing a Mattermost plugin. Feel free to base your own plugin off this repository.
 
 ## Getting Started
-Shallow clone the repository to a directory matching your plugin name:
+Shallow clone the repository to a directory outside of `$GOPATH` matching your plugin name:
```

Comment:
Is it necessary? It seems just works well in GOPATH with go1.12, or `GO111MODULE=on` flag with go1.11+

Reply:
If you just use the makefile, this isn't necessary. But if you can't run even a simple `go test` without manually setting `GO111MODULE=on` if the repo is in GOPATH.
But you're right, this will be no issue with g1.12.

What is the advantage of keeping this in GOPATH?

Reply:
Nothing about additional advantage. I agree to move go module, It doesn't have to be GOPATH, but also it should not be restriction about GOPATH (like "outside of `$GOPATH`") because many Gophers still have GOPATH.

Personally, I already have all my projects in the GOPATH... Haha.

Reply:
I don't have a strong opinion on moving away from `GOPATH`. Let's see what the other people think.

Reply:
Using the force here, but I'd propose not directing where the code should live at all, and relying on `GO111MODULE` to get the semantics we want "for free". I feel like this will result in fewer surprises for plugin developers who might not be familiar with the ecosystem changes that have been happening.

Reply:
> I feel like this will result in fewer surprises for plugin developers who might not be familiar with the ecosystem changes that have been happening.

I accentually thought the opposite. If you place the code outside of `GOPATH` you will have the "old behaviour" of everything working fine. But if you place it inside `GOPATH` commands like `go test` won't work without `GO111MODULE=on`. There seams to me a lot of possibilities for surprises.

Reply:
Ah, right -- this is only a problem if we also remove vending, since otherwise the non-module-aware tooling should work just fine.